packages/core/entity/FileExplorerURL/index.ts

```diff
@@ -157,4 +157,79 @@ export default class FileExplorerURL {
+            urlComponents.hierarchy
+                ?.map((annotation) => this.convertGroupByToPython(annotation))
+                .join("") || "";
+
+        // Group filters by name and use OR to concatenate same filter values
+        const filterGroups = new Map();
+        urlComponents.filters?.forEach((filter) => {
+            const pythonQueryString = filterGroups.get(filter.name);
+            if (!pythonQueryString) {
+                filterGroups.set(filter.name, this.convertFilterToPython(filter));
+            } else {
+                filterGroups.set(
+                    filter.name,
+                    pythonQueryString.concat(` | ${this.convertFilterToPython(filter)}`)
+                );
+            }
+        });
+
+        // Chain the filters together
+        let filterQueryString = "";
+        filterGroups.forEach((value) => {
+            filterQueryString = filterQueryString.concat(`.query('${value}')`);
+        });
+
+        const sortQueryString = urlComponents.sortColumn
+            ? this.convertSortToPython(urlComponents.sortColumn)
+            : "";
+        // const fuzzy = [] // TO DO: support fuzzy filtering
+
+        const hasQueryElements = groupByQueryString || filterQueryString || sortQueryString;
+        const imports = "import pandas\n";
+        const comment = hasQueryElements ? "#Query on dataframe df" : "#No options selected";
+        const fullQueryString = `${comment}${
+            hasQueryElements && `\ndf${groupByQueryString}${filterQueryString}${sortQueryString}`
+        }`;
+        return `${imports}${collectionString}${fullQueryString}`;
+    }
+
+    private static convertSortToPython(sortColumn: FileSort) {
+        return `.sort_values(by='${sortColumn.annotationName}', ascending=${
+            sortColumn.order == "ASC" ? "True" : "False"
+        })`;
+    }
+
+    private static convertGroupByToPython(annotation: string) {
+        return `.groupby('${annotation}', group_keys=True).apply(lambda x: x)`;
+    }
+
+    private static convertFilterToPython(filter: FileFilter) {
+        // TO DO: Support querying non-string types
+        if (filter.value.includes("RANGE")) {
+            return;
+            //     let begin, end;
+            //     return `\`${filter.name}\`>="${begin}"&\`${filter.name}\`<"${end}"`
+        }
+        return `\`${filter.name}\`=="${filter.value}"`;
+    }
+
+    private static convertCollectionToPython(collection: Collection | undefined) {
+        if (collection?.uri) {
+            const comment = "#Convert current datasource file to a pandas dataframe";
+            const extension = collection.uri.substring(collection.uri.lastIndexOf(".") + 1);
+            // Currently suggest setting all fields to strings; otherwise pandas assumes type conversions
+            // TO DO: Address different non-string type conversions
+            const code = `df = pandas.read_${extension}('${collection.uri}').astype('str')`;
```

Comment:
windows will need to denote the url as raw 
`df = pandas.read_csv(r'C:\Users\Brian\Desktop\Variance Paper Dataset.csv').astype('str')
`

Reply:
Added OS check in [9b14578](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/98/commits/9b14578d20d689b15227e9c03bd0adba2f3fbf5e) and [afb7f3c](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/98/commits/afb7f3c5be3ed9974ad5d539e4aac3e5de823ce2)